carla_controller/simulation.py

```diff
@@ -1,19 +1,57 @@
+from weather import random_weather
+
 import carla
 
+from pathlib import Path
 from random import choice
 from time import sleep
+from typing import Optional
+from uuid import uuid4
+
+
+PEDESTRIAN_CROSSING_DEFAULT = 0.15
+DEFAULT_VEHICLES = 50
+DEFAULT_PEDESTRIANS = 25
+CAMERA_START_DELAY = 5
+CAMERA_SHOT_DELAY = 1.0
+MAP_LOAD_DELAY = 2.0
+DEFAULT_OUTPUT_DIRECTORY = "./output/"
 
 class Simulation():
 
 
-    def __init__(self):
+    def __init__(
+        self,
+        run_id : Optional[str] = None,
+        map : Optional[str] = None,
+        output_directory : Optional[str] = None,
+        vehicles : int = DEFAULT_VEHICLES,
+        pedestrians : int = DEFAULT_PEDESTRIANS,
+        pedestrian_cross_percentage : float = PEDESTRIAN_CROSSING_DEFAULT,
+        hero_spawn_point = None,
+        weather_params = None
+    ):
+        if run_id is None:
+            run_id = uuid4()
+        self.run_id = run_id
+
         # Create our client
         self.client = carla.Client('localhost', 2000)
         self.client.set_timeout(5.0)
 
         # Create our world object
         self.world = self.client.get_world()
 
+        if map is None:
+            map = choice(self.client.get_available_maps())
+        self.client.load_world(map)
+        # Give the map a moment to load
+        sleep(MAP_LOAD_DELAY)
+
+        if weather_params == None:
+            weather_params = random_weather()
```

Comment:
If we want to skew weather a bit, we can control each element directly with:
```
weather = carla.WeatherParameters(
    cloudyness=80.0,
    precipitation=30.0,
    sun_altitude_angle=70.0)
world.set_weather(weather)
```
and confirm with
```
print(world.get_weather())
```

Reply:
Also have a list of static presets, which is maybe what the random is using
https://carla.readthedocs.io/en/0.9.3/python_api/#carlaweatherparameters

Except that none of these options are night.

Reply:
I can make it night - frankly I'm happy you found that. I couldn't find this list earlier. I did find some examples of dynamic lighting but that's a lot of extra overhead for something that doesn't really help us. I'll rework it to just utilize the random weather and try to incorporate random night better.